### ISSUE TITLE
PN532 with advanced series (large TFT)

### ISSUE THREAD

**Issue by johnvcoleman** [OpenEVSE/openevse_esp32_firmware#900]
I’ve just installed a PN532 board, connecting it to the qwiic connector on the back of the TFT. The board powers up, but there doesn’t appear to be any communication. 

1. On power up the PN532 board gave a red status light.
2. I went into the UI and turned on the RFID setting. EVSE status lights turned purple.
3. Tapped the scan button, it began counting down, but no amount of card waving produced a read.

I tried several RFID cards and double checked my wiring, no joy.

Can anyone spare some troubleshooting tips or links to troubleshooting guides?

The PN532 board:
https://www.amazon.com/gp/product/B01I1J17LC/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

I Attached the data export and an image. Yes, I know, I was very careful taking the picture and had adult supervision. :)

Thank you!

![IMG_3411](https://github.com/user-attachments/assets/3c291633-62be-4c2d-a96b-f639c683949c)

[RFID-export.json](https://github.com/user-attachments/files/16822770/RFID-export.json)

**Comment by clowrey** [comment]
Did you check the address switch on the board? It has to be in the right position to match whatever address is set in the firmware. I have that same module it works, but I forget what position the switches have to be in.

**Comment by johnvcoleman** [comment]
The silkscreen next to the switches list the settings as HSU, I2C, and SPI. I have them set to I2C. (admittedly, that was a guess)

**Comment by chris1howell** [comment]
I2C is correct. Make sure SCL and SDA are correct. The pin out on the
OpenEVSE side is standard Sparkfun Quiic.

**Comment by clowrey** [comment]
Dont know then.. Its possible the RFID board is defective or something else, those cheap modules are rarely even tested after SMD assembly so some have faults.. But I cant say much else from here, better pictures might help ;) It does look like your wiring is correct.

**Comment by clowrey** [comment]
Actually I think your DIP switch is in the wrong position, unless you adjusted it after the picture was taken. It should have one ON and one OFF
![image](https://github.com/user-attachments/assets/baf0cdde-632f-4de5-995d-800a55ea76f1)
![image](https://github.com/user-attachments/assets/faf329e9-a495-4cb1-83e9-6d1a9f640829)

**Comment by johnvcoleman** [comment]
Good eye, I spotted the switch settings after taking that photo. Sorry. I’ll double check the wires again. I guess it’s also possible I got a DOA board.

**Comment by johnvcoleman** [comment]
Ok, got a new board. (At least Amazon is fast.) Same behavior, board powers up (using i2c switch settings) but does not appear to read or communicate.  Then I remembered that I've been running a dev build from March (I updated to the dev build to fix a TFT issue) Fearing that not being on the most recent firmware was causing the issue with the reader I've spent the last hour or so trying to update the firmware, which consistently hangs around 98% and never completes. I Reset the device, completed the wizard, attempted the firmware update again (both release and dev) and it just hangs. (I waited 15 minutes before giving up) It looks like I may need to get the serial programmer and do this the hard way.

<img width="657" alt="Screenshot 2024-09-02 at 2 13 57 PM" src="https://github.com/user-attachments/assets/873954ea-c59c-448d-bc30-ed4ef2db0774">
<img width="886" alt="Screenshot 2024-09-02 at 2 13 12 PM" src="https://github.com/user-attachments/assets/a9f981de-747b-441c-9e19-a0d72c1c444f">

**Comment by clowrey** [comment]
Yes make a serial adapter like this. For the price the LCD board should definitely come with USB port IMO. 
![PXL_20240902_205328649 (Medium)](https://github.com/user-attachments/assets/60204827-71af-4b08-8d33-962f6a4eeb70)

**Comment by chris1howell** [comment]
The partitions for the OS and Over the air update need to be increased in size. 

The repartition process for the TFT LCD in the short term requires a USB => Serial adapter and esptools (requires Python and pyserial). 

Download the 5.1.2 - openevse_wifi_tft_v1.bin, bootloader_16mb.bin and partitions_16mb.bin from

https://github.com/OpenEVSE/openevse_esp32_firmware/releases/tag/v5.1.2

Connect the module to the USB => Serial adapter, then put the module in bootloader mode by holding BOOT and then pressing RESET then release BOOT

Run the command:

esptool --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_size=detect 0x1000 bootloader_16mb.bin 0x8000 partitions_16mb.bin 0x10000 openevse_wifi_tft_v1.bin

**Comment by rbray89** [comment]
![Image](https://github.com/user-attachments/assets/f4faec7a-af01-4abe-ab36-ba25ef6c720b)
![Image](https://github.com/user-attachments/assets/f22100ab-8708-4dfb-a94c-999b838b1081)

Looks like your connections are fine. Make sure the DIP switch is set correctly. Note that the reader likes to read in the orientation I have mine in, with the traces towards the RFID chips. I'm running the latest build now.
